app/controllers/devise_sessions_controller.rb

```diff
@@ -82,6 +90,10 @@ def check_login_state
     redirect_to new_user_session_path unless login_state
   end
 
+  def check_password_ok
+    redirect_to user_session_path(login_state_id: params[:login_state_id]) unless login_state.password_ok == true
```

Comment:
This can just be unless login_state.password_ok

Reply:
Ah, yes, of course!  Removed the unnecessary `== true` now.